pubsub/gossipsub/gossipsub-v1.0-old.md

```diff
@@ -0,0 +1,413 @@
+[@jamesray1], [@vasco-santos]
+
+[@whyrusleeping]: https://github.com/whyrusleeping
+[@yusefnapora]: https://github.com/yusefnapora
+[@raulk]: https://github.com/raulk
+[@vyzo]: https://github.com/vyzo
+[@Stebalien]: https://github.com/Stebalien
+[@jamesray1]: https://github.com/jamesray1
+[@vasco-santos]: https://github.com/vasco-santos
+
+See the [lifecycle document][lifecycle-spec] for context about maturity level
+and spec status.
+
+[lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
+
+---
+
+This is the specification for an extensible baseline pubsub protocol,
+based on randomized topic meshes and gossip. It is a general purpose
+pubsub protocol with moderate amplification factors and good scaling
+properties. The protocol is designed to be extensible by more
+specialized routers, which may add protocol messages and gossip in
+order to provide behaviour optimized for specific application
+profiles.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Contents**
+
+- [gossipsub: An extensible baseline pubsub protocol](#gossipsub-an-extensible-baseline-pubsub-protocol)
+  - [Implementation status](#implementation-status)
+  - [In the beginning was floodsub](#in-the-beginning-was-floodsub)
+    - [Ambient Peer Discovery](#ambient-peer-discovery)
+    - [Flood routing](#flood-routing)
+    - [Retrospective](#retrospective)
+  - [Controlling the flood](#controlling-the-flood)
+    - [randomsub: A random message router](#randomsub-a-random-message-router)
+    - [meshsub: An overlay mesh router](#meshsub-an-overlay-mesh-router)
+    - [gossipsub: The gossiping mesh router](#gossipsub-the-gossiping-mesh-router)
+  - [The gossipsub protocol](#the-gossipsub-protocol)
+    - [Control messages](#control-messages)
+    - [Router state](#router-state)
+    - [Topic membership](#topic-membership)
+    - [Message processing](#message-processing)
+    - [Heartbeat](#heartbeat)
+    - [Control message piggybacking](#control-message-piggybacking)
+    - [Protobuf](#protobuf)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Implementation status
+
+- Go: [libp2p/go-libp2p-pubsub/gossipsub.go](https://github.com/libp2p/go-libp2p-pubsub/blob/master/gossipsub.go) (experimental)
+- JS: [ChainSafeSystems/gossipsub-js](https://github.com/ChainSafeSystems/gossipsub-js) – work in progress; check branches and PRs.
+- Rust: [libp2p/rust-libp2p#898](https://github.com/libp2p/rust-libp2p/pull/898) implements the spec but is missing some features. [libp2p/rust-libp2p#767](https://github.com/libp2p/rust-libp2p/pull/767) is an alternative, partial implementation that differs slightly from the spec (see [#142](https://github.com/libp2p/specs/issues/142) for details).
+- Gerbil: [vyzo/gerbil-simsub](https://github.com/vyzo/gerbil-simsub) (simulator)
```

Comment:
```suggestion
```

Already appears on the entrypoint doc. Worth putting here as well?

Reply:
no, forgot to remove it from the old file.